src/power_grid_model/validation/_validation.py

```diff
@@ -449,6 +423,39 @@ def is_valid_component(data, component):
     return results
 
 
+def _validate_required_power_sigma_or_p_q_sigma(
+    data: SingleDataset,
+    sensor: ComponentType,
+) -> list[MissingValueError]:
+    """
+    Check that either `p_sigma` and `q_sigma` are all provided, or that `power_sigma` is provided.
+
+    Args:
+        data: SingleDataset, pgm data
+        sensor: only of types ComponentType.sym_power_sensor or ComponentType.asym_power_sensor
+    """
+    result: list[MissingValueError] = []
+
+    if sensor in data:
+        sensor_data = data[sensor]
+        p_sigma = sensor_data["p_sigma"]
+        q_sigma = sensor_data["q_sigma"]
+
+        asym_axes = tuple(range(sensor_data.ndim, p_sigma.ndim))
+        all_pq_sigma_missing_mask = np.all(np.isnan(p_sigma), axis=asym_axes) & np.all(
+            np.isnan(q_sigma), axis=asym_axes
+        )
+
+        result += _validate_required_in_data(
+            {sensor: sensor_data[all_pq_sigma_missing_mask]}, required={sensor: ["power_sigma"]}
+        )
+        result += _validate_required_in_data(
+            {sensor: sensor_data[~all_pq_sigma_missing_mask]}, required={sensor: ["p_sigma", "q_sigma"]}
+        )
```

Comment:
This check extends the original amount of errors (see the additions to `test_validation_functions.py`)

TBD: we can also remove this step and only keep the `power_sigma` one to keep the original behavior